### PR TITLE
Adds EmaCrossFuturesFrontMonthAlgorithm

### PR DESCRIPTION
**Pull request by AlexCatarino**
#### Description
This example shows how to create an EMA cross algorithm for a futures' front contract. Once the contract is added, the indicators are registered to a new consolidator and warmed up with historical data. When a contract is removed, the consolidator is removed and the indicators are reset. We don't need to liquidate it, because it's liquidated automatically since it has expired.

#### Motivation and Context
Provide an example for common user requests. 

#### Requires Documentation Change
Yes. I can be featured at [Futures' section](https://www.quantconnect.com/docs/data-library/futures).

#### How Has This Been Tested?
Locally and in QuantConnect Cloud using 20190101-2020603 date range.

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:
- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Algorithm.CSharp/EmaCrossFuturesFrontMonthAlgorithm.cs [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        public override void Initialize()
+        {
+            SetStartDate(2013, 10, 08);
+            SetEndDate(2013, 10, 10);
+            SetCash(1000000);
+
+            var future = AddFuture(Futures.Metals.Gold);
+
+            // Only consider the front month contract
+            future.SetFilter(x => x.FrontMonth().OnlyApplyFilterAtMarketOpen());
```
Comment (Martin-Molinero) [inline Algorithm.CSharp/EmaCrossFuturesFrontMonthAlgorithm.cs:49]: Might be worth adding a comment why is `OnlyApplyFilterAtMarketOpen` used? For performance I presume

File: Algorithm.Python/EmaCrossFuturesFrontMonthAlgorithm.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        self.consolidator = self.ResolveConsolidator(self.symbol, Resolution.Minute)
+        self.RegisterIndicator(self.symbol, self.fast, self.consolidator)
+        self.RegisterIndicator(self.symbol, self.slow, self.consolidator)
+
+        # Update the consolidator with historical data
+        # This will update the registered indicators
+        history = self.History(self.symbol, self.slow.WarmUpPeriod)
+        for index, row in history.iterrows():
+            time = index[2]
+            bar = TradeBar(time, self.symbol, row.open, row.high, row.low, row.close, 0)
```
Comment (Martin-Molinero) [inline Algorithm.Python/EmaCrossFuturesFrontMonthAlgorithm.py:97]: We could also use the WarmUpIndicator() helper method API?

File: Algorithm.Python/QuantConnect.Algorithm.PythonTools.pyproj [hunk 3]
```diff
@@ -133,7 +134,6 @@
     <Compile Include="PearsonCorrelationPairsTradingAlphaModelFrameworkAlgorithm.py" />
     <Compile Include="PortfolioRebalanceOnCustomFuncRegressionAlgorithm.py" />
     <Compile Include="PortfolioRebalanceOnDateRulesRegressionAlgorithm.py" />
-    <Compile Include="PsychSignalSentimentRegressionAlgorithm.py" />
```
Comment (Martin-Molinero) [inline Algorithm.Python/QuantConnect.Algorithm.PythonTools.pyproj:136]: 👍

File: Algorithm.Python/EmaCrossFuturesFrontMonthAlgorithm.py (hunk 2 above)
Comment (AlexCatarino) [reply Algorithm.Python/EmaCrossFuturesFrontMonthAlgorithm.py:97]: Great idea. It's particular useful in Python to avoid pandas conversion.

**Comment by AlexCatarino** [comment]
> Couple of 0 points in the charts

I have tested the algorithm with `Futures.Indices.SP500EMini` and 2019010-20200605 date range to evaluate whether it was an issue with `WarmUpIndicator()` or a data one. 

We can rule out an issue with `WarmUpIndicator()` because there is only one zero data-point in the chart that corresponds to Jan 1st, 2019. It doesn't happen on Jan 1st, 2020, because there is no security change on that day.

The zero data-point on that day is caused by the missing holiday in the MHDB that is reported in #3461.

**Pull request merged**